### PR TITLE
Get cluster status API

### PR DESCRIPTION
**Pull request by tock-ibm**
Add an API that allows any user to detect the configured nodes from ClusterConfig.Nodes
and the status of the nodes: who is the leader, and which nodes are active.

Signed-off-by: Yoav Tock <tock@il.ibm.com>

### CODE REVIEW

File: internal/bcdb/transaction_processor_test.go [hunk 1]
```diff
@@ -565,6 +569,25 @@ func TestTransactionProcessor(t *testing.T) {
...
+	t.Run("get cluster status", func(t *testing.T) {
+		cryptoDir, conf := testConfiguration(t)
+		require.NotEqual(t, "", cryptoDir)
+		defer os.RemoveAll(conf.LocalConfig.Server.Database.LedgerDirectory)
+		env := newTxProcessorTestEnv(t, cryptoDir, conf)
+		defer env.cleanup()
+
+		setupTxProcessor(t, env, worldstate.DefaultDBName)
+
+		require.Eventually(t, func() bool {
```
Comment (gennadylaventman) [inline internal/bcdb/transaction_processor_test.go:582]: Isn't it possible that leader will be elected fast enough and this condition never become true?

**Review by gennadylaventman — COMMENTED** [review]
Some minor comments

File: pkg/constants/http.go [hunk 2]
```diff
@@ -45,12 +45,14 @@ const (
...
-	GetConfig          = "/config/tx"
-	GetNodesConfig     = "/config/node"
-	GetNodeConfig      = "/config/node/{nodeId}"
+	ConfigEndpoint    = "/config/"
+	PostConfigTx      = "/config/tx"
+	GetConfig         = "/config/tx"
+	GetNodeConfigPath = "/config/node"
+	GetNodeConfig     = "/config/node/{nodeId}"
 	GetLastConfigBlock = "/config/block/last"
+	GetClusterStatus  = "/config/cluster"
```
[the remark this answers is not in the document]
Comment (tock-ibm) [reply pkg/constants/http.go:54]: done

File: internal/bcdb/db.go [hunk 3]
```diff
@@ -388,6 +402,56 @@ func (d *db) GetConfigBlock(querierUserID string, blockNumber uint64) (*types.Ge
...
+		Nodes:   nodes,
+		Version: configResponse.GetMetadata().GetVersion(),
+	}
+
+	leader, active := d.txProcessor.ClusterStatus()
+
+	// The configResponse is from the world-state-db, whereas the (leader, active) pair is from the block-replicator.
+	// Since they may be out of sync (if the request hits right in the middle of a config change), we make sure the
+	// response is self-consistent.
+	for _, activeID := range active {
```
[the remark this answers is not in the document]
Comment (tock-ibm) [reply internal/bcdb/db.go:422]: Hmm... I prefer not to mess with it now for saving a single loop... so let's leave it.

File: internal/bcdb/transaction_processor_test.go (hunk 1 above)
Comment (tock-ibm) [reply internal/bcdb/transaction_processor_test.go:582]: This condition asserts that the current node is the leader; that is when IsLeader() returns nil, the current node is leader, so it is fine.

Comment (gennadylaventman) [reply internal/bcdb/transaction_processor_test.go:581]: I see

**Review by gennadylaventman — APPROVED** [review]

**Pull request merged**